public/app/plugins/panel/geomap/layers/basemaps/default.ts

```diff
@@ -0,0 +1,38 @@
+import { MapLayerRegistryItem, MapLayerOptions, GrafanaTheme2 } from '@grafana/data';
+import Map from 'ol/Map';
+import { carto } from './carto';
+import { esriXYZTiles } from './esri';
+import { xyzTiles } from './generic';
+import { standard } from './osm';
+import { config } from 'app/core/config';
+
+// Array of base map options to search through
+const baseLayers = [carto, esriXYZTiles, xyzTiles, standard];
+
+// Configured base map from the server side
+const defaultBaseLayer = config.geomapDefaultBaseLayer;
+
+// Default base layer depending on the server setting
+export const defaultLayers: MapLayerRegistryItem<any> = {
+  id: 'default',
+  name: 'Default base layer',
+  isBaseMap: true,
+
+  create: (map: Map, options: MapLayerOptions, theme: GrafanaTheme2) => ({
+    init: () => {
```

Comment:
```suggestion
  create: (map: Map, options: MapLayerOptions, theme: GrafanaTheme2) => {
      let layer = carto;
      if(config.geomapDefaultBaseLayer) {
          layer = baseLayers.find((baseLayer) => baseLayer.id === defaultBaseLayer.type);
          if(!layer) {
              ??? what should we do?
          }
      }
      return layer.create(map,options,theme);
  }
```

Reply:
I like this approach!  but rather than implement the functions directly, return the whole handler.  This will make the "update" function work also

Reply:
Makes sense. updated it! :)